app/models/user.rb

```diff
@@ -107,6 +107,11 @@ def inactive_message
     "Sorry, this account has been disabled."
   end
 
+  # Returns all users who match the query.
+  def self.search_from_query(members, query)
+    User.enabled.where.not(id: members).where(arel_table[:username].matches(query)).pluck(:username)
```

Comment:
Option 2 and 3 are exactly the same. A Scope is just a function returning a ActiveRelation Object, it's just a different syntax and it's less obvious if you write it in the 2. way.